posthog/warehouse/api/external_data_source.py

```diff
@@ -834,6 +873,15 @@ def database_schema(self, request: Request, *arg: Any, **kwargs: Any):
             ]
 
             return Response(status=status.HTTP_200_OK, data=result_mapped_to_options)
+        elif source_type == ExternalDataSource.Type.CHARGEBEE:
+            api_key = request.data.get("api_key", "")
+            site_name = request.data.get("site_name", "")
+
+            if not validate_chargebee_credentials(api_key=api_key, site_name=site_name):
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={"message": "Invalid credentials: Chargebee credentials are incorrect"},
+                )
```

Comment:
Nice, we also wanna check the subdomain to ensure users can't do any RCE attacks here, see https://github.com/PostHog/posthog/blob/master/posthog/warehouse/api/external_data_source.py#L781-L786

Reply:
ah yes, I missed that, thanks!